src/D2L.CodeStyle.Analyzers/Immutability/ImmutabilityAnalyzer.cs

```diff
@@ -23,7 +23,9 @@ public sealed class ImmutabilityAnalyzer : DiagnosticAnalyzer {
 			Diagnostics.UnnecessaryMutabilityAnnotation,
 			Diagnostics.UnexpectedConditionalImmutability,
 			Diagnostics.ConflictingImmutability,
-			Diagnostics.InvalidAuditType
+			Diagnostics.InvalidAuditType,
+
+			Diagnostics.MissingTransitiveImmutableAttribute

```

Comment:
Nit: remove extra space
```suggestion
			Diagnostics.InvalidAuditType,
			Diagnostics.MissingTransitiveImmutableAttribute

```

Reply:
I explicitly separated it since it is a somewhat separate diagnostic from the rest, in terms of scope. Didn't come up with a category name to comment on the groups though. The first ones are ~all about actual type immutability, and this one (and maybe additional related) is more about consistent contracts.

Still think I should collapse it?

Reply:
If you separated them for a reason then I think that's fine. It might still be worth adding a comment there to explain why that one is separate.